pkg/controller/atlasdeployment/serverless_deployment.go

```diff
@@ -54,3 +77,16 @@ func ensureServerlessInstanceState(ctx *workflow.Context, project *mdbv1.AtlasPr
 		return atlasDeployment, workflow.Terminate(workflow.Internal, fmt.Sprintf("unknown deployment state %q", atlasDeployment.StateName))
 	}
 }
+
+func isEqual(a *mongodbatlas.Cluster, c *mongodbatlas.Cluster) bool {
+	// TODO: include ServerlessBackupOptions and TerminationProtectionEnabled
+	if len(*a.Tags) == len(*c.Tags) {
+		for i, aTags := range *a.Tags {
```

Comment:
This will find differences if the arrays contain the same key-value pairs but in different order.
One way to avoid this is to order both arrays before comparing.
Another is to put all keys into a map from the first array and check all key pairs from the second array are in the map built before.